heat/core/logical.py

```diff
@@ -106,7 +106,10 @@ def allclose(x, y, rtol=1e-05, atol=1e-08, equal_nan=False):
     """
     if np.isscalar(x):
         try:
-            x = factories.array([float(x)])
+            if isinstance(y, dndarray.DNDarray):
+                x = factories.array([float(x)], device=y.device)
+            else:
+                x = factories.array([float(x)])
```

Comment:
this should still get a device. maybe make a device parameter above it to set the devices of both `x` and `y` to be the same

Reply:
If no device is given (i.e. device=None), they have the same default device. In this case, I wanted to change the device implicitly, if a scalar is compared with a tensor.